### PR TITLE
Update deps, use edition 2018

### PR DESCRIPTION
**Pull request by najamelan**
This is a mindless chore PR:

- update to edition 2018
- update serde_cbor to 0.10

----
I ended up doing this because I was having a look to see what it would take to port this to futures-codec to frame AsyncRead/AsyncWrite from futures 0.3. It turned out that after this PR, it literally just takes changing:

`use tokio_io::codec::{Decoder as IoDecoder, Encoder as IoEncoder};`

into:

`use futures_codec::{Decoder as IoDecoder, Encoder as IoEncoder};`

At least is compiles and the tests run. I'll go and play with it a bit!

### CODE REVIEW

File: src/lib.rs
Comment (vorner) [inline src/lib.rs]: Nitpick: this could be folded into a single line, eg` use serde_cbor::ser::{IoWrite, Serializer}`.

**Review by vorner — APPROVED** [review]
Thank you for the pull request :-).

There's this small nitpick, I'll leave it up to you if you want to adopt it or not. I'll do a new release after it gets merged.

File: src/lib.rs
Comment (najamelan) [reply src/lib.rs]: I always leave "Allow edits from maintainers" checked, so don't hesitate! But I just updated it for you.

Comment (vorner) [reply src/lib.rs]: Thank you :-)

I prefer not to use that, it's better for author to acknowledge it and maybe learn something new, if it wasn't just oversight.

**Pull request merged**
